src/main/java/unicash/ui/StatusBarFooter.java

```diff
@@ -1,29 +1,106 @@
 package unicash.ui;
 
+import static unicash.ui.StyleSheet.TEXT_FILL_BLACK;
+import static unicash.ui.StyleSheet.TEXT_FILL_GREEN;
+import static unicash.ui.StyleSheet.TEXT_FILL_RED;
+
 import java.nio.file.Path;
 import java.nio.file.Paths;
 
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
 import javafx.fxml.FXML;
 import javafx.scene.control.Label;
 import javafx.scene.layout.Region;
+import unicash.model.transaction.Transaction;
 
 /**
- * A ui for the status bar that is displayed at the footer of the application.
+ * A UI Controller class for the status bar that is displayed at the
+ * footer of the application. Contains information about data path and
+ * balance indicator.
  */
 public class StatusBarFooter extends UiPart<Region> {
 
     private static final String FXML = "StatusBarFooter.fxml";
+    private static final String BALANCE_FORMAT = "Rolling Balance: $%.2f";
```

Comment:
Maybe make this a global string inside our `utils` package. This way if we do change the decimal formatting, we don't have to edit different places and potentially miss one out

Reply:
Thinking about it, it actually doesn't make sense for the balance formatter to be here at all. I've actually gone ahead and removed this entirely, and `StatusBarFooter` now relies on `Amount`'s String representation alone. I think this makes more sense as `Amount` has its own currency symbol and precision formatting, any indicator or label pertaining to amounts should not override whatever is specified by `Amount`.